### PR TITLE
Support for order by

### LINKED ISSUE #89: Add support for ORDER BY

### PR DESCRIPTION
**Pull request by patofp**
This PR add support for queries with order by like this one:

```
PREFIX foaf:    <http://xmlns.com/foaf/0.1/>

SELECT ?name
WHERE { ?x foaf:name ?name ; foaf:age ?age }
ORDER BY DESC(?name) ?age DESC(?age) ASC(?name) DESC((isBlank(?x) || isBlank(?age)))
```

Where variables, functions and binary operations can be defined as conditions.

Closes #89

### CODE REVIEW

**Review by geoffrey-pitman — APPROVED** [review]
👍

File: modules/engine/src/main/scala/com/gsk/kg/engine/analyzer/FindUnboundVariables.scala [hunk 1]
```diff
@@ -85,6 +85,7 @@ object FindUnboundVariables {
       case Limit(limit, r)         => r.pure[ST]
       case Distinct(r)             => r.pure[ST]
       case Group(vars, func, r)    => r.pure[ST]
+      case DAGOrder(variable, r)   => r.pure[ST]
```
Comment (pepegar) [inline modules/engine/src/main/scala/com/gsk/kg/engine/analyzer/FindUnboundVariables.scala:88]: Shouldn't we be checking this `variable` to see if it's bound or not?

File: modules/parser/src/test/scala/com/gsk/kg/sparqlparser/OrderConditionParserSpec.scala [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+//(triple ?x <http://xmlns.com/foaf/0.1/name> ?name)
+//(triple ?x <http://xmlns.com/foaf/0.1/empId> ?emp)
+//)))
+//
+//(project (?name)
+//(order (?name ?emp (desc ?emp) (asc ?name) (desc (|| (isBlank ?x) (isBlank ?emp))))
+//(bgp
+//(triple ?x <http://xmlns.com/foaf/0.1/name> ?name)
+//(triple ?x <http://xmlns.com/foaf/0.1/empId> ?emp)
+//)))
```
Comment (pepegar) [inline modules/parser/src/test/scala/com/gsk/kg/sparqlparser/OrderConditionParserSpec.scala:115]:
```suggestion
```

**Review by pepegar — APPROVED** [review]
jUst added a couple of comments, but this looks great!

File: modules/engine/src/main/scala/com/gsk/kg/engine/analyzer/FindUnboundVariables.scala (hunk 1 above)
Comment (patofp) [reply modules/engine/src/main/scala/com/gsk/kg/engine/analyzer/FindUnboundVariables.scala:88]: oh! that name is incorrect as that node is a list of conditions:

```
  @Lenses final case class Order[A](conds: NonEmptyList[ConditionOrder], r: A)
      extends DAG[A]
```

And ConditionOrder:

```
sealed trait ConditionOrder extends Expression
object ConditionOrder {
  final case class ASC(e: Expression)  extends ConditionOrder
  final case class DESC(e: Expression) extends ConditionOrder
}
```

So with that `e: Expression` I think it is already being checked whether are variables bounded. Is that correct? Although I'm going to check it.

File: modules/parser/src/test/scala/com/gsk/kg/sparqlparser/OrderConditionParserSpec.scala (hunk 2 above)
Comment (patofp) [reply modules/parser/src/test/scala/com/gsk/kg/sparqlparser/OrderConditionParserSpec.scala:115]: upssss! 😅

File: modules/engine/src/main/scala/com/gsk/kg/engine/analyzer/FindUnboundVariables.scala (hunk 1 above)
Comment (patofp) [reply modules/engine/src/main/scala/com/gsk/kg/engine/analyzer/FindUnboundVariables.scala]: Opened new issue to check variables on some nodes: #313

**Pull request merged**